components/delete_post_modal/delete_post_modal.jsx

```diff
@@ -34,16 +33,30 @@ export default class DeletePostModal extends React.PureComponent {
     handleDelete = async () => {
         const {
             actions,
-            channelName,
-            focusedPostId,
             post,
-            teamName,
         } = this.props;
 
+        let permalinkPostId = '';
+
         const result = await actions.deleteAndRemovePost(post);
 
-        if (post.id === focusedPostId && channelName) {
-            browserHistory.push('/' + teamName + '/channels/' + channelName);
+        const matchUrlForDMGM = matchPath(this.props.location.pathname, {
+            path: '/:teamName/messages/:username/:postid',
```

Comment:
Note: Change this to be based on a constant so we don't mess this up when someone changes the format of url's

Reply:
On second thought, i kept the constasts just in the same file. Constants file is huge and adding them there when not using them anywhere else as of now did not make a whole lot of sense